db/version_edit_test.cc

```diff
@@ -309,6 +309,109 @@ TEST_F(VersionEditTest, BlobFileAdditionAndGarbage) {
   TestEncodeDecode(edit);
 }
 
+TEST_F(VersionEditTest, WAL) {
+  VersionEdit edit;
+
+  for (uint64_t log_number = 1; log_number <= 20; log_number++) {
+    switch (log_number % 3) {
+      case 0:
+        edit.AddWal(AddedWal(log_number, rand() % 100));
+        break;
+      case 1:
+        edit.ArchiveWal(log_number);
+        break;
+      case 2:
+        edit.DeleteWal(DeletedWal(log_number, rand() % 2 == 0));
+        break;
+    }
+  }
+
+  TestEncodeDecode(edit);
+}
+
+TEST_F(VersionEditTest, AddedWal) {
+  constexpr uint64_t kLogNumber = 10;
+  constexpr uint64_t kBytes = 100;
+
+  VersionEdit edit;
+  edit.AddWal(AddedWal(kLogNumber, kBytes));
+
+  auto wals = edit.GetNewWals();
```

Comment:
`const auto& wals` ? Also, `const AddedWal& wal` below. (Affects the other two test cases as well.)